src/main/java/com/google/cloud/spanner/myadapter/statements/SessionStatementParser.java

```diff
@@ -0,0 +1,285 @@
+import com.google.cloud.spanner.myadapter.session.SessionState;
+import com.google.cloud.spanner.myadapter.session.SessionState.SessionVariableScope;
+import com.google.cloud.spanner.myadapter.session.SystemVariable;
+import com.google.cloud.spanner.myadapter.statements.SimpleParser.TableOrIndexName;
+import com.google.common.collect.ImmutableList;
+import java.util.ArrayList;
+import javax.annotation.Nullable;
+
+/** Simple parser for session management commands (SET/SHOW/RESET variable_name) */
+@InternalApi
+public class SessionStatementParser {
+  public abstract static class SessionStatement {
+    public abstract StatementResult execute(SessionState sessionState);
+  }
+
+  static class VariableColumn {
+    static class Builder {
+      Builder columnName(String columnName) {
+        this.columnName = columnName;
+        return this;
+      }
+
+      Builder variableName(String variableName) {
+        this.variableName = variableName;
+        return this;
+      }
+
+      Builder setGlobal() {
+        this.scope = SessionVariableScope.GLOBAL;
+        return this;
+      }
+
+      Builder setSystemVariable() {
+        this.systemVariable = true;
+        return this;
+      }
+
+      VariableColumn build() {
+        return new VariableColumn(this);
+      }
+
+      String columnName;
+      String variableName;
+      SessionVariableScope scope = SessionVariableScope.SESSION;
+      boolean systemVariable = false;
+    }
+
+    String columnName;
+    String variableName;
+    SessionVariableScope scope;
+    boolean systemVariable;
+
+    VariableColumn(Builder builder) {
+      this.columnName = builder.columnName;
+      ;
+      this.variableName = builder.variableName;
+      this.scope = builder.scope;
+      this.systemVariable = builder.systemVariable;
+    }
+  }
+
+  static class SetStatement extends SessionStatement {
+    static class Builder {
+      boolean systemVariable = false;
+      SessionVariableScope scope = SessionVariableScope.SESSION;
+      String name = null;
+      String value = null;
+
+      Builder systemVariable() {
+        this.systemVariable = true;
+        return this;
+      }
+
+      Builder global() {
+        this.scope = SessionVariableScope.GLOBAL;
+        return this;
+      }
+
+      Builder name(String name) {
+        this.name = name;
+        return this;
+      }
+
+      Builder value(String value) {
+        this.value = value;
+        return this;
+      }
+
+      SetStatement build() {
+        return new SetStatement(this);
+      }
+    }
+
+    boolean systemVariable;
+    SessionVariableScope scope;
+    String name;
+    String value;
+
+    SetStatement(Builder builder) {
+      this.systemVariable = builder.systemVariable;
+      this.scope = builder.scope;
+      this.name = builder.name;
+      this.value = builder.value;
+    }
+
+    @Override
+    public StatementResult execute(SessionState sessionState) {
+      sessionState.set(name, value, scope);
```

Comment:
Consider binging the autocommit logic here. This seems a more reasonable place. Although make sure that Spanner is updated first.